Mintos.lua

```diff
@@ -5,104 +5,151 @@ WebBanking {
+        local code = credentials[1]
+
+        connection:request("POST",
+        "https://www.mintos.com/en/login/twofactor",
+        "_one_time_password=" .. code .. "&_csrf_token=" .. csrfToken,
+        "application/x-www-form-urlencoded; charset=UTF-8")
+
+        if string.match(connection:getBaseURL(), "login") then
+            return LoginFailed
+        end
     end
 end
 
 function ListAccounts (knownAccounts)
     local html = HTML(connection:get("https://www.mintos.com/en/my-settings/"))
     local accountNumber = html:xpath("//*/table[contains(concat(' ', normalize-space(@class), ' '), ' js-investor-settings ')]/tr[1]/td[@class='data']"):text()
 
-    local accounts = {}
+    return {
+        {
+            name = "Available Funds",
+            accountNumber = accountNumber .. "-1",
+            currency = "EUR",
+            type = AccountTypeGiro
+        }, {
+            name = "Invested Funds",
+            accountNumber = accountNumber .. "-2",
+            currency = "EUR",
+            portfolio = true,
+            type = AccountTypePortfolio
+        }
+    }
+end
+
+function ParseStatements (receivedStatements)
+    if #receivedStatements > 0 then
+        for i, element in ipairs(receivedStatements) do
+            local transaction = ParseStatements(element)
+
+            local dateString = element["date"]
+            local day, month, year = dateString:match(datePattern)
+
+            local purpose = element["details"]
+            local amount = element["turnover"]
+
+            local transaction = {
+                bookingDate = os.time({day=day,month=month,year=year,hour=0,min=0}),
+                purpose = StripHtml(purpose),
+                amount = tonumber(amount)
+            }
 
-    table.insert(accounts, {
-        name = 'Available Funds',
-        accountNumber = accountNumber .. '-1',
-        currency = "EUR",
-        type = AccountTypeGiro
-    })
+            table.insert(transactions, transaction)
+        end
+    end
+end
 
-    table.insert(accounts, {
-        name = 'Invested Funds',
-        accountNumber = accountNumber .. '-2',
-        currency = "EUR",
-        portfolio = true,
-        type = AccountTypePortfolio
-    })
+function GetStatementsForPage (since, page)
+    local list = JSON(connection:request("POST",
+    "https://www.mintos.com/en/account-statement/page/",
+    "account_statement_filter[fromDate]=" .. os.date("%d.%m.%Y", since) .. "&account_statement_filter[toDate]=" .. os.date("%d.%m.%Y", os.time()) .. "&account_statement_filter[maxResults]=".. maxStatementsOnPage .. "&account_statement_filter[currency]=978&account_statement_filter[page]=" .. page,
+    "application/x-www-form-urlencoded; charset=UTF-8",
+    {
+        ["x-requested-with"] = "XMLHttpRequest"
+    })):dictionary()
+
+    return list["data"]["accountStatements"]
 
-    return accounts
 end
 
 function RefreshAccount (account, since)
```

Comment:
This function is a little long, maybe break it down at the if/else?

Reply:
... also the condition doesn't really reveal itself to me. Maybe also put it in a function with a good name that returns a boolean? Are there booleans in Lua? I have no idea ... 😂

Reply:
Good point. Should be more readable now